### ISSUE TITLE
CFDoc comments for functions

### ISSUE THREAD

**Issue by shaedrich** [KamasamaK/vscode-cfml#5]
Hi KamasamaK,
let's say I have the following:
```
/**
* @property required numeric a - an awesome number
*/
```
Then the plugin recognizes `numeric` and `a` properly. But it does not get the param description and the required keyword. I tried out many commonly known syntaxes but nothing worked. Did I something wrong or did I found a bug or simply a not yet implemented feature?

Best regards

**Comment by KamasamaK** [comment]
Can you please include what is being documented underneath it as well? I am not seeing the issue and am confused by the redundant information.

**Comment by shaedrich** [comment]
What is redundant in my information?

The comment could be longer but that does not matter. If we just had 
```
/**
* Formats duration
* moment.utc(moment.duration(duration).as('milliseconds')).format('HH:mm')
* @property numeric duration - duration in minutes
* @return duration in hh:mm
* @output false
* @ReturnType string
* @access private
*/
function formatDuration(required numeric duration) {
	return "#(duration \ 60)# hours, #(duration % 60)# minutes";
}
```

it would only display the following (_No argument description_):

![capturfiles](https://user-images.githubusercontent.com/11225821/33671206-4a2cfe7e-daa7-11e7-9448-b10675350255.png)
![capturfiles_1](https://user-images.githubusercontent.com/11225821/33671207-4a469f64-daa7-11e7-8bee-e5a469a52fab.png)

**Comment by KamasamaK** [comment]
There is nothing called `property` that would do anything here. For an argument hint, you want to use the name of the argument -- `@duration` or `@duration.hint`. You can find more information from [Adobe](https://helpx.adobe.com/coldfusion/developing-applications/the-cfml-programming-language/extending-coldfusion-pages-with-cfml-scripting/defining-components-and-functions-in-cfscript.html), [Adam Cameron](https://github.com/adamcameron/cfscript/blob/master/cfscript.md#functionargument-annotations), or [Ben Nadel](https://www.bennadel.com/blog/1662-learning-coldfusion-9-cfscript-updates-for-coldfusion-components.htm). 

FYI, it is not good practice to include anything but documentation. You should rather declare the function with its `access` and `returnType` as:
```cfml
private string function formatDuration(required numeric duration) {}
```

**Comment by shaedrich** [comment]
So, CFDoc comments differ from JavaDoc when declaring an argument...

But
```
/**
* @duration duration in minutes
* or
* @duration.hint duration in minutes
*/
```
does not work either.
How should I declare a parameter description the way you did it?

**Comment by KamasamaK** [comment]
Yes, it differs from JavaDoc style. I'm not sure what you mean that it doesn't work. It looks fine to me.

![image](https://user-images.githubusercontent.com/22951521/33674749-68a8f354-da7e-11e7-91fe-f528f4f792c6.png)

![image](https://user-images.githubusercontent.com/22951521/33674783-84e5909a-da7e-11e7-9027-422e0cabaddd.png)

**Comment by shaedrich** [comment]
I did it like you but I still get "_No argument description_":

![capturfiles_2](https://user-images.githubusercontent.com/11225821/33675268-49da4db8-dab2-11e7-80a1-b4128f60e713.png)

**Comment by KamasamaK** [comment]
Have you saved the file recently? I notice the function hint is also out of date and saving will force it to parse again.

**Comment by shaedrich** [comment]
I did. So this is a bug?

**Comment by KamasamaK** [comment]
Perhaps, but it doesn't seem to be one I can reproduce.

**Comment by shaedrich** [comment]
Hmmm... Sound's strange... But that's what bugs do to our developer brains from time to time, right?^^

**Comment by KamasamaK** [comment]
Based on the spacing around the argument, it appears that you have an older version. Please restart VS Code and be sure you have the latest version of VS Code and the extension.

**Comment by shaedrich** [comment]
Yeah, I currently have some problems with updating VS Code properly...
